geonode/layers/api/serializers.py

```diff
@@ -71,6 +71,59 @@ class Meta:
     attribute = serializers.CharField(read_only=True)
 
 
+class FeatureInfoTemplateField(DynamicComputedField):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_attribute(self, instance):
+        if instance.use_featureinfo_custom_template and instance.featureinfo_custom_template:
+            return instance.featureinfo_custom_template
+        else:
+            _attributes = instance.attributes.filter(visible=True).order_by('display_order')
+            if _attributes.count():
```

Comment:
it would be better using the `.exists()` instead of the `.count()`

Reply:
wrong, we want to print out a template only if there's at least one attribute.

Reply:
updated due to https://stackoverflow.com/a/23585057/7597536